addons/partner_autocomplete/models/res_company.py

```diff
@@ -12,3 +16,57 @@ class ResCompany(models.Model):
     def _inverse_partner_gid(self):
         for company in self:
             company.partner_id.partner_gid = company.partner_gid
+
+    def _enrich(self):
+        """
+        This method calls the partner autocomplete service from IAP to enrich the partner related fields
+        of the company.
+        """
+        self.ensure_one()
+
+        company_domain = self._get_company_domain()
+        if not company_domain:
+            return {'error': _("Could not extract company domain. Impossible to enrich the company.")}
+
+        company_data = self.env['res.partner'].enrich_company(company_domain, False, self.vat)
+        if company_data.get('error'):
+            return {'error': company_data.get('error_message', _('Unkown reason'))}
+
+        for field in ['state_id', 'country_id']:
+            company_data[field] = company_data.get(field, {}).get('id')
+
+        # Keep only truthy values that are not already set on the target partner
+        company_data = {field: value for field, value in company_data.items()
+                        if field in self.partner_id._fields and value and not self.partner_id[field]}
+
+        additional_data = company_data.pop('additional_info', False)
```

Comment:
My bad, the process before where you remove fields that are not in the fields definition of "res.partner" will remove "additional_info", right?
So you should pop it a bit earlier.

Reply:
Helloo, my bad, I didn't double check this one